cmd/ipfs/main.go

```diff
@@ -195,21 +194,90 @@ func checkDebug(req *cmds.Request) {
 	}
 }
 
+func apiAddrOption(req *cmds.Request) (ma.Multiaddr, error) {
+	apiAddrStr, apiSpecified := req.Options[corecmds.ApiOption].(string)
+	if !apiSpecified {
+		return nil, nil
+	}
+	return ma.NewMultiaddr(apiAddrStr)
+}
+
 func makeExecutor(req *cmds.Request, env interface{}) (cmds.Executor, error) {
+	exe := cmds.NewExecutor(req.Root)
+	cctx := env.(*oldcmds.Context)
 	details := commandDetails(req.Path)
-	client, err := commandShouldRunOnDaemon(*details, req, env.(*oldcmds.Context))
+
+	// Check if the command is disabled.
+	if details.cannotRunOnClient && details.cannotRunOnDaemon {
+		return nil, fmt.Errorf("command disabled: %v", req.Path)
+	}
+
+	// Can we just run this locally?
+	if !details.cannotRunOnClient && details.doesNotUseRepo {
+		return exe, nil
+	}
+
+	// Get the API option from the commandline.
+	apiAddr, err := apiAddrOption(req)
 	if err != nil {
 		return nil, err
 	}
 
-	var exctr cmds.Executor
-	if client != nil && !req.Command.External {
-		exctr = client.(cmds.Executor)
-	} else {
-		exctr = cmds.NewExecutor(req.Root)
+	// Force the daemon when the API flag is passed (unless we're trying to
+	// _run_ the daemon).
+	daemonForced := apiAddr != nil && req.Command != daemonCmd
```

Comment:
pedantic nits:
>Force the daemon when the API flag is passed 

Should probably be more specific here, forcing the daemon to what?
Initialize temporarily?

 `daemonForced` -> `daemonRequested`
Despite a user supplying an API addr, commands don't aways force the daemon to do anything.
It's requested, and can be denied if the request is incompatible.

Reply:
Thanks, updated. I went back and forth on that several times but decided that e.g., `ipfs version` should just be run on the client regardless.